tuning/utils/preprocessing_utils.py

```diff
@@ -364,7 +371,18 @@ def get_jsonl_object():
                     output_field_name: data[output_field_name],
                 }
 
-    return Dataset.from_generator(get_jsonl_object)
+    def get_json_object():
```

Comment:
Potentially can refactor in to the same method....
```py
   def get_json_object():
        with open(data_path, "r", encoding="utf-8") as json_file:
            if file_extension == ".jsonl":
                data_stream = [json.loads(line) for line in jsonl_file]
            elif file_extension == ".json":
                data_stream = json.load(json_file)
            for data in data_stream:
                yield {
                    input_field_name: data[input_field_name],
                    output_field_name: data[output_field_name],
                }
```

is `file_extension` accessible in the method given its a function defined inside a function?

Reply:
Even nicer! `file_extension` is not accessible in the method, but it could be something like this:
```
    def get_json_object():
        with open(data_path, "r", encoding="utf-8") as json_file:
            file_extension = os.path.splitext(data_path)[-1].lower()
            if file_extension == ".jsonl":
                data_stream = (json.loads(line) for line in json_file)
            elif file_extension == ".json":
                data_stream = json.load(json_file)
            else:
                raise ValueError("Unsupported file format! Use 'json' or 'jsonl'.")
            for data in data_stream:
                yield {
                    input_field_name: data[input_field_name],
                    output_field_name: data[output_field_name],
                }

    return Dataset.from_generator(get_json_object)
```